File: chatdownloader/src/metrics/bits.py
"""
The bits metrics
"""
from _types import Comment

from .abstractmetric import AbstractMetric

WEIGHT_BITS = 0.02


class Bits(AbstractMetric):
    """
    Bits metric
    """

    @classmethod
    def can_parallelize(cls) -> bool:
        return True

    @classmethod
    def get_name(cls) -> str:
        return 'bits'

    def get_metric(self, comment: Comment, sequence_no: int \
                   ) -> list[dict[str, int]]:
        return self._shortcut_for_this_comment_user(
            comment, comment.message.bits_spent * WEIGHT_BITS)

File: chatdownloader/src/metrics/bits.rs
/*
The bits metric
*/
use crate::_types::twitchtypes::Comment;
use crate::metrics::metrictrait::AbstractMetric;

use std::collections::HashMap;

const WEIGHT_BITS: f32 = 0.1;

#[allow(dead_code)]
#[derive(Default, Debug)]
pub struct Bits;

impl AbstractMetric for Bits {
    async fn new() -> Self {
        Self
    }

    fn can_parallelize(&self) -> bool {
        true
    }

    fn get_name(&self) -> String {
        String::from("bits")
    }

    fn get_metric(&mut self, comment: Comment, _sequence_no: u32) -> HashMap<String, f32> {
        let score = comment.message.bits_spent.unwrap_or(0) as f32 * WEIGHT_BITS;
        self._shortcut_for_this_comment_user(comment, score)
    }
}